.github/workflows/publish.yml

```diff
@@ -0,0 +1,27 @@
+name: Publish
+on:
+  push:
+    branches: [master, "@cbrown/743-mkdocs-site"]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Build
+        run: |
+          cp -r resources docs/img/resources && mkdocs build -c -f mkdocs.yml
```

Comment:
What do you think about including a call to `./autogen.sh` to create all the various `*.md` at publish time instead of storing them under source control?

Reply:
Good idea! I will add this to the github action

Reply:
@bdsoha addressed this in the most recent few commits to this branch. Github action installs dependencies for `autogen.sh` and runs it (see https://github.com/Collinbrown95/diagrams/blob/%40cbrown/743-mkdocs-site/.github/workflows/publish.yml#L16-L36). Also removed `docs/img/resources` folder and `docs/nodes` folder as these are now auto generated in the CI pipeline.